h2/src/main/org/h2/mvstore/tx/Transaction.java

```diff
@@ -300,24 +337,85 @@ public long setSavepoint() {
     }
 
     /**
-     * Mark an entry into a new SQL statement execution within this transaction.
-     * @param maps set of maps used by statement about to be executed
+     * Returns whether statement dependencies are currently set.
+     *
+     * @return whether statement dependencies are currently set
      */
-    public void markStatementStart(Set<MVMap<?, ?>> maps) {
+    public boolean hasStatementDependencies() {
+        return !mapRoots.isEmpty();
+    }
+
+    /**
+     * Mark an entry into a new SQL statement execution within this transaction.
+     *
+     * @param isolationLevel
+     *            isolation level
+     * @param currentMaps
+     *            set of maps used by statement about to be executed
+     * @param allMaps
+     *            set of all maps within transaction, may be modified by this
+     *            method
+     */
+    public void markStatementStart(IsolationLevel isolationLevel, HashSet<MVMap<?, ?>> currentMaps,
```

Comment:
That design looks a little strange. Can transaction isolation level be changed after transaction was created?
I think level should be taken from session at transaction creation / first command execution time only.

Reply:
No, it can't. Do you prefer a separate method to change the level of transaction?

Reply:
Such method probably will be required for correct reimplementation of `SET TRANSACTION ISOLATION LEVEL` in the future, so I'll add it now.